dataflow/core.py

```diff
@@ -479,16 +479,32 @@ class DataType(object):
     *id* : string
         Name of the data type.
 
-    *cls* : Classs
+    *cls* : Class
         Python class defining the data type.
+
+    *export_types*: list(strings)
+        A list of implemented export types for the data type (binary, text, etc..)
+
     """
-    def __init__(self, id, cls):
+    def __init__(self, id, cls, export_types=None):
         self.id = id
         self.cls = cls
+        # export_types is e.g. {"columns": {"method_name": "to_column_text", "exporter": exporters.text}}
+        self.export_types = self._find_exporters()
 
     def get_definition(self):
-        return {"id": self.id}
-
+        return {"id": self.id, "export_types": list(self.export_types.keys())}
+
+    def _find_exporters(self):
+        exporters = {}
+        for n in dir(self.cls):
+            p = getattr(self.cls, n)
+            if type(p) == types.FunctionType:
+                export_name = getattr(p, 'export_name', 'default')
+                exporter = getattr(p, 'exporter', None)
+                if exporter is not None:
+                    exporters[export_name] = {"exporter": exporter, "method_name": p.__name__}
```

Comment:
We kind of control the decorators in this case, but if n is always right then yes we should use that instead.  I think export_types gets sent to the client, so plain dict was preferred.  That code could be adjusted if you want though, at the point where it gets sent (I think it's in get_instruments).

Reply:
You can't be sending `exporters[export_name]` to the client because it contains a python function handle.  I'll leave it as is anyway, but I changed __name__ to n.